recipes/vulkan-loader/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+import os
+import glob
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+
+
+class VulkanLoaderConan(ConanFile):
+    name = "vulkan-loader"
+    description = "Khronos official Vulkan ICD desktop loader for Windows, Linux, and MacOS."
+    topics = ("conan", "vulkan", "loader", "desktop", "gpu")
+    homepage = "https://github.com/KhronosGroup/Vulkan-Loader"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_wsi_xcb": [True, False],
+        "with_wsi_xlib": [True, False],
+        "with_wsi_wayland": [True, False],
+        "with_wsi_directfb": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "with_wsi_xcb": True,
+        "with_wsi_xlib": True,
+        "with_wsi_wayland": True,
+        "with_wsi_directfb": False,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "pkg_config"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.with_wsi_xcb
+            del self.options.with_wsi_xlib
+            del self.options.with_wsi_wayland
+            del self.options.with_wsi_directfb
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.get_safe("with_wsi_directfb"):
+            # TODO: directfb package
+            raise ConanInvalidConfiguration("Conan recipe for DirectFB is not available yet.")
+        if not tools.is_apple_os(self.settings.os) and not self.options.shared:
+            raise ConanInvalidConfiguration("Static builds are not supported on {}".format(self.settings.os))
+
+    def requirements(self):
+        self.requires("vulkan-headers/{}".format(self.version))
+        if self.options.get_safe("with_wsi_xcb") or self.options.get_safe("with_wsi_xlib"):
+            self.requires("xorg/system")
+        if self.options.get_safe("with_wsi_wayland"):
+            self.requires("wayland/1.18.0")
+
+    def package_id(self):
+        if self.settings.compiler == "Visual Studio":
```

Comment:
Why is this?
This will alias static and dynamic runtime.

Reply:
vulkan-loader always link to static runtime

Reply:
fixed

Reply:
Static runtime is often chosen to make distribution easier.
Thanks for fixing it.

Reply:
Not sure I want to spend time with a feature that maintainers don't want to support: https://github.com/KhronosGroup/Vulkan-Loader/issues/343

Reply:
I tried:
- remove hardcoded SHARED in add_library
- remove .def if static
- add cfgmgr32 in system_libs for windows

test_package fails at runtime

Reply:
So static fails on Windows?
Does static work on Linux?

Reply:
It fails at runtime on Windows. Didn't try Linux.

Reply:
The reason it fails for static on windows, is because the initialization of the library does not happen.
See https://github.com/KhronosGroup/Vulkan-Loader/blob/aed40d52837dba3850a27c61529de6fa25979d19/loader/loader.c#L7877-L7893
There is no way to call these functions as vulkan-loader does not have any headers.

The lines below are for gcc and clang.
I don't know why they disallow a static linux executable as the lines look fine to me.

But hey, it's working.
Thanks @SpaceIm!